x-pack/plugins/security_solution/public/resolver/store/data/reducer.test.ts

```diff
@@ -48,5 +90,127 @@ describe('Resolver Data Middleware', () => {
     it('should indicate there are additional children', () => {
       expect(selectors.hasMoreChildren(store.getState())).toBe(true);
     });
+    it('should have the correct related events', () => {
+      const selectedEventsByEntityId = selectors.relatedEventsByEntityId(store.getState());
+      const selectedEventsForFirstChildNode = selectedEventsByEntityId.get(firstChildNodeInTree.id)!
+        .events;
+
+      expect(selectedEventsForFirstChildNode).toBe(firstChildNodeInTree.relatedEvents);
+    });
+    it('should indicate the correct related event count for each category', () => {
+      const selectedRelatedInfo = selectors.relatedEventInfoByEntityId(store.getState());
+      const displayCountsForCategory = selectedRelatedInfo.get(firstChildNodeInTree.id)
+        ?.getNumberActuallyDisplayedForCategory!;
+      for (const typeCounted of Object.keys(statsForFirstChild)) {
+        expect(`${typeCounted}:${displayCountsForCategory(typeCounted)}`).toBe(
+          `${typeCounted}:${statsForFirstChild[typeCounted]}`
+        );
+      }
+    });
+    it('should not indicate the limit has been exceeded when the actual count >= stats count', () => {
+      const selectedRelatedInfo = selectors.relatedEventInfoByEntityId(store.getState());
+      const shouldShowLimit = selectedRelatedInfo.get(firstChildNodeInTree.id)
+        ?.shouldShowLimitForCategory!;
+      for (const typeCounted of Object.keys(statsForFirstChild)) {
+        expect(shouldShowLimit(typeCounted)).toBe(false);
+      }
+    });
+    it('should not indicate that there are any related events missing when the actual count >= stats count', () => {
+      const selectedRelatedInfo = selectors.relatedEventInfoByEntityId(store.getState());
+      const notDisplayed = selectedRelatedInfo.get(firstChildNodeInTree.id)
+        ?.getNumberNotDisplayedForCategory!;
+      for (const typeCounted of Object.keys(statsForFirstChild)) {
+        expect(notDisplayed(typeCounted)).toBe(0);
+      }
+    });
+  });
+
+  describe('when data was received and stats show more related than the API can provide', () => {
+    let firstChildNodeInTree: TreeNode;
+    let statsForFirstChild: Record<string, number>;
+    let categoryToOverCount: string;
+    beforeEach(() => {
+      const generator = new EndpointDocGenerator('seed');
```

Comment:
From a glance this seems similar to the `beforeEach` above. If that's the case, can you either:

* move duplicated logic into a `beforeEach` that runs before both sets of specs

or

* move duplicate logic into functions that can be called from each `beforeEach`. 

I think either approach would help readability and maintainability.

Reply:
https://github.com/elastic/kibana/pull/70097/commits/ddbc8f82b354287545d706445cc46c3a716a6fec